File: src/main/java/com/wafflestudio/draft/security/oauth2/client/TestOAuth2Client.java
package com.wafflestudio.draft.security.oauth2.client;

import org.springframework.stereotype.Component;

@Component
public class TestOAuth2Client implements OAuth2Client {

    static final public String OAUTH_TOKEN_PREFIX = "TEST";

    @Override
    public OAuth2Response userInfo(String accessToken) {
        return new OAuth2Response("TEST", "testuser@test.com", "testprofile/urls");
    }
}

File: src/main/java/com/wafflestudio/draft/security/oauth2/client/TestOAuth2Client.kt
package com.wafflestudio.draft.security.oauth2.client

import org.springframework.http.HttpStatus
import org.springframework.stereotype.Component

@Component
class TestOAuth2Client : OAuth2Client {
    override fun userInfo(accessToken: String?): OAuth2Response {
        return OAuth2Response("TEST", "$accessToken@test.com", HttpStatus.OK)
    }

    companion object {
        const val OAUTH_TOKEN_PREFIX = "TEST"
    }
}